src/button-angular/lib/google-pay-button.component.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ButtonManager, Config } from '@google-pay/lib/button-manager';
+import { Directive, ElementRef, Input, OnChanges, OnInit, Renderer2, SimpleChanges } from '@angular/core';
+import { name as softwareId, version as softwareVersion } from '../package.json';
+import { debounce } from '@google-pay/lib/debounce';
+
+@Directive({
+  selector: 'google-pay-button',
```

Comment:
Element selectors are usually encouraged for components rather than directives. Is there a particular reason you decided to go with a directive instead of a component?

Reply:
The reason was that this component didn't have a `template`. The Google Pay button is added directly to the DOM.

How do you think I should handle this?

Reply:
IMO you're doing the right thing here since it looks like you're just trying to match your existing custom element API.

Reply:
Thanks for the feedback.